util.go

```diff
@@ -79,7 +88,14 @@ func isStatus(statusCode int, responseBody string) (bool, error) {
 	}
 
 	if responsePayload.StatusCode == statusCode {
-		return true, nil
+		if reason == "" {
+			// We're not looking for a specific reason, so we're done
+			return true, nil
+		}
+		// Match end of status text to ": " + <reason>
+		if strings.HasSuffix(responsePayload.StatusText, ": "+reason) {
```

Comment:
I would like to split `responsePayload.StatusText` by `:` and see if everything after the colon is exactly the same as `reason`.